resolveImports.mjs

```diff
@@ -1,7 +1,19 @@
+// @ts-check
+
 import { parse } from "es-module-lexer";
 import path from "node:path";
 import { readFile, access } from "node:fs/promises";
 
+/**
+ * Resolved imports.
+ * @typedef {string[] | undefined} ResolvedImports
+ */
+
+/**
+ * Checks if a file exists.
+ * @param {string} module The path to the file.
```

Comment:
Probably a good idea to rename the arg `module` to something like `filePath` since the function is usable for any file, not just modules. Also, the type could probably also accept a file `URL` instance or whatever else the underlying Node.js APIs accept.